### PR TITLE
(#43) Analytics: Other Search Results Click

### LINKED ISSUE #43: Story: Analytics: Other: Search Result Clicks
## As a analytics person, I want to capture information about search result clicks, so that we can see what users click on

> **ESTIMATE** NN

### Acceptance Criteria

```gherkin
 Scenario: A user clicks on a search result on the first page
    Given the user navigates to "/?swKeyword=tumor"
     When the system displays "Results 1-20 of 11189 for: " "tumor" as an "h4" tag
      And the user clicks on the second result
     Then there should be an analytics event with the following details
      | key              | value                                                              | 
      | type             | Other                                                              | 
      | event            | SitewideSearchApp:Other:ResultClick                                | 
      | linkName         | SiteWideSearchResults                                              | 
      | data.resultIndex | 2                                                                  | 
      | data.resultUrl   | https://www.cancer.gov/types/pancreatic/patient/pnet-treatment-pdq | 
```

```gherkin
  Scenario: A user clicks on a search result on the second page
    Given the user navigates to "/?swKeyword=tumor&page=2&pageunit=20"
     When the system displays "Results 21-40 of 11189 for: " "tumor" as an "h4" tag
      And the user clicks on the second result
     Then there should be an analytics event with the following details
      | key              | value                                                                          | 
      | type             | Other                                                                          | 
      | event            | SitewideSearchApp:Other:ResultClick                                            | 
      | linkName         | SiteWideSearchResults                                                          | 
      | data.resultIndex | (int)22                                                                        | 
      | data.resultUrl   | https://www.cancer.gov/pediatric-adult-rare-tumor/rare-tumors/rare-bone-tumors | 
```

### Notes
* **The item index is based on its position in the overall result set, NOT its position on the page.**
* NOTE: on a search result page the page.additionalDetails will have the following items that can be referenced in launch for this click event...
   * page.additionalDetails.numberResults
   * page.additionalDetails.searchKeyword
   * page.additionalDetails.itemsPerPage
   * page.additionalDetails.pageNum

### PR DESCRIPTION
**Pull request by seyilonge-nci**
Closes #43

* Added tracking event and click handler
* Added integration tests
* Updated unit tests

### CODE REVIEW

**Comment by seyilonge-nci** [comment]
This would fail the build process till PR #47 gets merged as the integration test relies on the correct title for the item on the second page which is currently off.

File: src/components/molecules/search-results-list/results-list-item.jsx [hunk 1]
```diff
@@ -30,9 +35,24 @@ const ResultsListItem = ({ result = {}, language = 'en' }) => {
...
 		}
 	}
+
+	const handleResultItemTitleClick = () => {
+		const pageUnit = urlQuery.get('pageunit');
+		tracking.trackEvent({
+			type: 'Other',
+			event: 'SitewideSearchApp:Other:ResultClick',
+			linkName: 'SiteWideSearchResults',
+			resultIndex: +itemIndex + 1 + +pageUnit,
```
Comment (bryanpizzillo) [inline src/components/molecules/search-results-list/results-list-item.jsx:45]: This calculation should be done outside the item in the results list.

File: src/components/molecules/search-results-list/results-list-item.jsx [hunk 2]
```diff
@@ -30,9 +35,24 @@ const ResultsListItem = ({ result = {}, language = 'en' }) => {
 			);
 		}
 	}
+
+	const handleResultItemTitleClick = () => {
+		const pageUnit = urlQuery.get('pageunit');
```
Comment (bryanpizzillo) [inline src/components/molecules/search-results-list/results-list-item.jsx:40]: You should never be referencing query params except in either the route or the main view. This does not lend itself to easy testing since you have to mock a full url just to test if this component can display.

File: src/components/molecules/search-results-list/results-list-item.jsx (hunk 1 above)
Comment (bryanpizzillo) [reply src/components/molecules/search-results-list/results-list-item.jsx:45]: Additionally, it is unclear as to what numbers are what here.

Comment (seyilonge-nci) [reply src/components/molecules/search-results-list/results-list-item.jsx]: Updated! Moved computation outside of component.

File: src/components/molecules/search-results-list/results-list-item.jsx (hunk 2 above)
Comment (seyilonge-nci) [reply src/components/molecules/search-results-list/results-list-item.jsx]: Removed!

File: src/components/molecules/search-results-list/search-results-list.jsx [hunk 3]
```diff
@@ -57,11 +57,23 @@ const SearchResultsList = ({
...
 	);
 
+	/**
+	 *	Computes index value of result item taking current items per page (pageunit) into consideration
+	 *
+	 *	@param {number} Result index in current response
+	 *	@returns {number} Computed result index number
+	 */
+	const getResultIndex = (index) => {
+		const page = urlQuery.get('pageunit') || 0;
```
Comment (bryanpizzillo) [inline src/components/molecules/search-results-list/search-results-list.jsx:67]: Once again, why is this here? Is this different from `resultsPerPage`??

Comment (seyilonge-nci) [reply src/components/molecules/search-results-list/search-results-list.jsx]: Updated as discussed. Using `resultsPerPage` and `currentPage`.
